crates/extensions/c8y_http_proxy/src/actor.rs

```diff
@@ -308,12 +306,24 @@ impl C8YHttpProxyActor {
             download_info.auth = Some(Auth::new_bearer(token.as_str()));
         }
 
-        // Download a file to tmp dir
-        let file_name = request.file_path.file_name().unwrap().to_str().unwrap();
-        let downloader: Downloader = Downloader::new_sm(file_name, &None, self.tmp_dir.clone());
+        // Download file to the target directory with a temp name
+        let file_name = nanoid!();
```

Comment:
@reubenmiller suggested that this name be something deterministic. I'm thinking of using the target file name itself suffixed with a `.tmp` so that it's easier to identify even if those stay around due to some crashes/bugs.